skyportal/handlers/api/source.py

```diff
@@ -305,11 +337,28 @@ def get(self, obj_id=None):
 
         source_list = []
         for source in query_results["sources"]:
-            source.comments = source.get_comments_owned_by(self.current_user)
+            comments = source.get_comments_owned_by(self.current_user)
+            source.comments = sorted(comments, key=lambda x: x.created_at, reverse=True)
+            source.classifications = source.get_classifications_owned_by(
+                self.current_user
+            )
             source_list.append(source.to_dict())
             source_list[-1]["last_detected"] = source.last_detected
             source_list[-1]["gal_lon"] = source.gal_lon_deg
             source_list[-1]["gal_lat"] = source.gal_lat_deg
+            source_list[-1]["groups"] = (
+                Group.query.filter(
+                    Group.id.in_(
+                        DBSession()
+                        .query(Source.group_id)
+                        .filter(Source.obj_id == source_list[-1]["id"])
+                    )
+                )
+                .filter(
+                    Group.id.in_([g.id for g in self.current_user.accessible_groups])
+                )
+                .all()
+            )
```

Comment:
```suggestion
            source_list[-1]["groups"] = DBSession().query(Group).join(Source).filter(Source.obj_id == source_list[-1]["id"], Group.id.in_(user_accessible_groups)).all()
```

Reply:
Good, and also I've changed a similar part on line 242 (same thing, only for single source).